app/scripts/directives/bindService.js

```diff
@@ -70,28 +73,46 @@ function BindService($filter,
     }
   };
 
-  ctrl.$onInit = function() {
-    ctrl.steps = [];
-    if (ctrl.target.kind === 'Instance') {
-      ctrl.steps.push({
-        id: 'applications',
-        label: 'Applications',
-        view: 'views/directives/bind-service/select-application.html'
-      });
-    }
-    else {
-      ctrl.steps.push({
-        id: 'services',
-        label: 'Services',
-        view: 'views/directives/bind-service/select-service.html'
-      });
-    }
-    ctrl.steps.push({
-      label: 'Results',
-      id: 'results',
-      view: 'views/directives/bind-service/results.html'
+  var showBind = function() {
+    ctrl.nextTitle = 'Bind';
+
+    validityWatcher = $scope.$watch("ctrl.selectionForm.$valid", function(isValid) {
+      ctrl.steps[0].valid = isValid;
     });
+  };
 
+  var showResults = function() {
+    if (validityWatcher) {
+      validityWatcher();
+      validityWatcher = undefined;
+    }
+    ctrl.nextTitle = "Close";
+    ctrl.wizardComplete = true;
+
+    ctrl.bindService();
+  };
+
+  ctrl.$onInit = function() {
+    ctrl.serviceSelection = {};
+    var formStepId = (ctrl.target.kind === 'Instance') ? 'applications' : 'services';
```

Comment:
Would be it be simpler to have one step ID and then just change the step indicator label? I don't see where the ID is checked anywhere.